public/docs/_examples/dependency-injection/dart/lib/car/car.dart

```diff
@@ -0,0 +1,32 @@
+// #docregion
+import 'package:angular2/core.dart';
+
+// #docregion engine
+class Engine {
+  var cylinders = 4;
+}
+// #enddocregion engine
+
+// #docregion tires
+class Tires {
+  var make = 'Flintstone';
+  var model = 'Square';
+}
+
+// #enddocregion tires
+@Injectable()
+class Car {
+  //#docregion car-ctor
+  Engine engine;
+  Tires tires;
+  var description = "DI";
+  Car(this.engine, this.tires);
+  // #enddocregion car-ctor
+
+  // Method using the engine and tires
+  drive() {
+    return '${ this . description} car with ' +
```

Comment:
I think you could shorten the returned string to:

```
'$description car with ${engine.cylinders} cylinders and ${tires.make} tires.'
```

You might still need to split it over 2 or 3 lines, though. I think dartfmt will do this for you.